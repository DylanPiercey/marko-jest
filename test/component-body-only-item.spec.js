const path = require('path');
const { initComponent, createTestSandbox } = require('../test-utils');

describe('body-only component', () => {
  const componentClass = initComponent(path.resolve(__dirname, './resources/body-only-item/index.marko'));

  let testSandbox;
  let component;

  beforeEach(() => {
    testSandbox = createTestSandbox();
  });

  afterEach(() => {
    testSandbox.reset();
  });

  describe('on rendering body-only mode', () => {
    beforeEach(async () => {
      component = await testSandbox.renderComponent(componentClass, { showSpan: true, text: 'test' });
    });

    it('should render correctly', () => {
      expect(component.els).toMatchSnapshot();
    });
  });

  describe('on rendering NON-body-only mode', () => {
    beforeEach(async () => {
      component = await testSandbox.renderComponent(componentClass, { showSpan: false, text: 'test' });
    });

    it('should update the element', () => {
      expect(component.els).toMatchSnapshot();
    });
  });
});
