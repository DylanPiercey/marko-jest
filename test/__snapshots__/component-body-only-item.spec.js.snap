// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`body-only component on rendering NON-body-only mode should update the element 1`] = `
Array [
  test,
]
`;

exports[`body-only component on rendering body-only mode should render correctly 1`] = `
Array [
  <span>
    test
  </span>,
]
`;
